### PR TITLE
[CInterop] Fix keyword mangling in metadata mode

### PR DESCRIPTION
**Pull request by sbogolepov**

### CODE REVIEW

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrTextEmitter.kt [hunk 1]
```diff
@@ -391,7 +391,7 @@ class StubIrTextEmitter(
     }
 
     private fun renderClassifierDeclaration(classifier: Classifier): String =
-            kotlinFile.declare(classifier)
+            kotlinFile.declare(classifier).asSimpleName()
```
Comment (SvyatoslavScherbina) [inline Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrTextEmitter.kt:394]: Are there other usages of `declare`?

File: backend.native/tests/interop/basics/mangling_keywords.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import kotlin.test.*
+import mangling_keywords.*
+
+fun main() {
+    // Check that all Kotlin keywords are imported and mangled.
+    assertEquals(`as`, "as")
```
Comment (SvyatoslavScherbina) [inline backend.native/tests/interop/basics/mangling_keywords.kt:6]: Arguments should probably be swapped.

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrTextEmitter.kt (hunk 1 above)
Comment (sbogolepov) [reply Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrTextEmitter.kt:394]: No. I'll try to gradually refactor `KotlinFile` later.

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/KotlinCodeModel.kt [hunk 3]
```diff
@@ -320,7 +320,7 @@ abstract class KotlinFile(
                 // TODO: using original global name should be preferred to importing the clashed name.
             } else {
                 declaredProperties.add(name)
-                name.asSimpleName()
+                name
```
Comment (knebekaizer) [inline Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/KotlinCodeModel.kt:323]: Have you ever consider different types for MangledName and non-mangled? Even if they have the same text representation (for most of cases I guess) they still have different usage areas.

Comment (sbogolepov) [reply Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/KotlinCodeModel.kt:323]: Yeah, it's may be a good idea to use `inline class` for such thing.

**Pull request merged**